### PR TITLE
fix: fix ImageDataset axis order and add tests

### PR DESCRIPTION
**Pull request by ganow**
## Problem

`ImageDataset` had three bugs, with no tests to catch them:

1. **Wrong axis order in returned image array**
   `__getitem__` returned images in HWC format `(H, W, C)`, but PyTorch's `DataLoader` and most deep learning models expect CHW format `(C, H, W)`. This caused shape mismatches when feeding images into networks.

2. **Incorrect auto-detection of stimulus names**
   When `stimulus_names=None`, file stems were extracted using a custom `_removesuffix` helper instead of `Path.stem`. This was unnecessarily verbose and fragile.

3. **Non-deterministic order of auto-detected stimulus names**
   When `stimulus_names=None`, `Path.glob()` was used without sorting, so the order of stimulus names was filesystem-dependent and not reproducible.

## Fix

- Added `.transpose(2, 0, 1)` to convert image arrays from HWC to CHW before returning.
- Replaced the `_removesuffix`-based list comprehension with `path.stem`.
- Added `sorted()` to ensure auto-detected stimulus names are always in alphabetical order.

## Tests

Added `tests/dl/torch/test_dataset.py` with 9 test cases:

- `test_getitem_returns_chw_shape` — verifies CHW axis order using a non-square image (H≠W) to fully discriminate every axis
- `test_getitem_preserves_channels` — verifies per-channel values are correctly mapped after transpose
- `test_dataloader_integration_batch_shape` — end-to-end check via `DataLoader` (the original failure path)
- `test_value_range_normalized_to_unit_interval` — verifies pixel values are in `[0, 1]`
- `test_len_matches_stimulus_names` — verifies `__len__`
- `test_explicit_stimulus_names_respected` — verifies explicit `stimulus_names` are used as-is
- `test_explicit_stimulus_names_preserve_input_order` — verifies input order is preserved when `stimulus_names` is given
- `test_auto_detected_stimulus_names_use_stem` — verifies file stems are used when `stimulus_names=None`
- `test_auto_detected_stimulus_names_are_sorted` — verifies alphabetical ordering of auto-detected names

Also removed the empty `TestImageDataset` stub from `test_torch.py`.

### CODE REVIEW

File: bdpy/dl/torch/dataset.py [hunk 1]
```diff
@@ -184,7 +181,7 @@ def __getitem__(self, index: int):
         stimulus_name = self._stimulus_names[index]
         image = Image.open(Path(self.root_path) / f"{stimulus_name}.{self._extension}")
         image = image.convert("RGB")
-        return np.array(image) / 255.0, stimulus_name
+        return np.array(image).transpose(0, 3, 1, 2) / 255.0, stimulus_name
```
Comment (ganow) [inline bdpy/dl/torch/dataset.py]: bugfix

```suggestion
        return np.array(image).transpose(2, 0, 1) / 255.0, stimulus_name
```

File: tests/dl/torch/test_dataset.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import tempfile
+import unittest
+from pathlib import Path
+
+import numpy as np
+import torch
```
Comment (KenyaOtsuka) [inline tests/dl/torch/test_dataset.py]: `torch` seems to be unused in this file. Could you remove it?

File: tests/dl/torch/test_dataset.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    Image.fromarray(data).save(path)
+
+
+class TestImageDataset(unittest.TestCase):
+    def setUp(self) -> None:
+        self.tmpdir = tempfile.TemporaryDirectory()
+        root = Path(self.tmpdir.name)
+        _save_image(root / "a.jpg", r=200, g=100, b=50)
+        _save_image(root / "b.jpg", r=10, g=20, b=30)
+        _save_image(root / "c.jpg", r=0, g=128, b=255)
```
Comment (KenyaOtsuka) [inline tests/dl/torch/test_dataset.py]: For tests that check exact pixel values, it may be better to use a lossless format such as PNG.

**Review by KenyaOtsuka — COMMENTED** [review]
Thanks for the fix. I left a few minor comments.

Optionally, since this changes the behavior of ImageDataset from HWC to CHW, it might be helpful to document that ImageDataset now returns images in CHW format.

File: tests/dl/torch/test_dataset.py (hunk 2 above)
Comment (ganow) [reply tests/dl/torch/test_dataset.py]: that's true. thank you for mentioning it

Comment (ganow) [reply tests/dl/torch/test_dataset.py]: fixed in c82dfac

File: tests/dl/torch/test_dataset.py (hunk 3 above)
Comment (ganow) [reply tests/dl/torch/test_dataset.py]: fixed in 57451e7

**Review by KenyaOtsuka — APPROVED** [review]
LGTM. Thank you!
